test/ng/directive/inputSpec.js

```diff
@@ -1196,7 +1196,7 @@ describe('input', function() {
 
       it('should validate if max is empty', function() {
         $rootScope.maxVal = undefined;
-        $rootScope.value = new Date(9999, 11, 31, 23, 59, 59);
+        $rootScope.value = new Date(3000, 11, 31, 23, 59, 59);
```

Comment:
It probably has something to do with how MS Edge handles UTC -> local datetime conversion while overflowing 9999. E.g. with my UTC+2 timezone, the UTC value `9999-12-31T21:59:59.999` (which corresponds to `9999-12-31T23:59:59.999` local time) is fine, but the UTC value `9999-12-31T22:00:00.000` (which corresponds to `+10000-01-01T00:00:00.000` local time) "confuses" MS Edge.

It may also have something to do with ISO 8601 defaulting to min. 4 digits per year, but _permitting the expansion of the year representation but only by prior agreement between the sender and the receiver_ (whatever that means).